### PR TITLE
Add first two models to allow trusts to be retrieved from API via search

### PR DESCRIPTION
**Pull request by reggieb**
### Context

Before being able to display Academy data in the front end, objects are required to manage retrieval of data from the API. This PR adds the first of those object - `Trust`.

As it is the first data retrieval object being created, a number of supporting gems were required so that it's behaviour could be easily tested.

### Changes proposed in this pull request

- Add `Trust` model - to manage trust data retrieval from API
- Add `BearerToken` model - to manage token retrieval - for API authentication
- Add Webmock to control and test calls to API within specs
- Add Faker to generate random test data
- Add FactoryBot to create populated objects easily for testing
- Add Faraday to simplify HTTP requests
- Add oauth2 gem to simplify retrieval of bearer token

### Guidance to review

With this change in place it is possible to retrieve trusts from the API within the Rails console with the command:
```ruby
trusts = Trust.search('Academy')
```
`trusts` will be an array of Trust instances populated with data from the API.

[Trello Ticket](https://trello.com/c/wUrP3hJ5/550-search-and-select-name-of-trust-outgoing-trust)

### CODE REVIEW

**Review by himal-mandalia — APPROVED** [review]
Very nice and clean

File: app/models/bearer_token.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  def json
+    @json ||= JSON.parse(response.body, symbolize_names: true)
+  end
+
+  def response
+    @response ||= Faraday.post(
+      SOURCE_URL,
+      grant_type: 'client_credentials',
+      scope: "api://6db0ed68-bd2e-455f-9a27-a24ed70dd6f8/.default",
```
Comment (himal-mandalia) [inline app/models/bearer_token.rb:20]: Should this be set somewhere rather than hardcoded?

Comment (reggieb) [reply app/models/bearer_token.rb]: I've moved the scope to the credentials.

**Review by himal-mandalia — APPROVED** [review]
Looks good!

**Pull request merged**